File: lib/prompts/templates/strategyList.ts
import type { Blueprint, GenerateRequest, SectionSpec } from '../../types'

export function buildPrompt(
  spec: SectionSpec,
  blueprint: Blueprint,
  req: GenerateRequest
): string {
  const { lesson_metadata } = req
  return `Generate a list of reading strategies for this study guide.

## Blueprint Context
- Lesson Title: ${blueprint.title}
- Core Concept: ${blueprint.core_concept}
- Subject: ${lesson_metadata.subject}
- Grade Level: ${lesson_metadata.grade_level}
- Market: ${lesson_metadata.market}

## Instructions
Generate 4–5 reading comprehension strategies relevant to ${lesson_metadata.subject} for Grade ${lesson_metadata.grade_level}. For each strategy:
1. Give it a clear, memorable name
2. Write a 1–2 sentence description of what the strategy involves and when to use it
3. List 3–5 signal words or phrases that indicate when this strategy is helpful

Strategies should be practical and appropriate for ${lesson_metadata.grade_level} students.

## Output Schema
{
  "strategies": [
    {
      "name": "string",
      "description": "string (1–2 sentences)",
      "signal_words": ["string"]
    }
  ]
}

Section spec: ${JSON.stringify(spec)}`
}

File: backend/study-guide-agent/app/prompts/templates/strategy_list.py
from __future__ import annotations

from app.types import Blueprint, GenerateRequest


def build_prompt(spec, blueprint: Blueprint, request: GenerateRequest) -> str:
    del spec

    strategy_list_schema = """{
    "title": "Strategy List",
    "strategies": [
        {
            "name": "string",
            "when_to_use": "string",
            "steps": [
                "string"
            ]
        }
    ]
}"""

    prompt_lines = [
        "Create the strategy list section for a K-12 study guide.",
        f"- Lesson title: {blueprint.title}",
        f"- Essential question: {blueprint.essential_question}",
        f"- Core concept: {blueprint.core_concept}",
        "Requirements:",
        "- Provide 3 practical reading or thinking strategies that help students succeed with this lesson.",
        "- Each strategy must include when_to_use and a short ordered steps list.",
        "- Keep the strategies directly tied to the lesson's core concept and purpose-identification work.",
        f"- Keep the reading level close to Grade {request.lesson_metadata.grade_level}.",
        "Expected JSON schema:",
        strategy_list_schema,
        "Return only JSON.",
    ]

    return "\n".join(prompt_lines)
